src/test/java/com/ibm/etcd/client/EtcdTestSuite.java

```diff
@@ -62,7 +63,8 @@ public static void tearDown() throws IOException {
 
     static void waitForStartup() throws Exception {
         if(etcdProcess == null) return;
-        TimeLimiter tl = SimpleTimeLimiter.create(Executors.newCachedThreadPool());
+        ExecutorService es = Executors.newSingleThreadExecutor();
+        TimeLimiter tl = SimpleTimeLimiter.create(es);
```

Comment:
@gi-wg2 could you put a try/finally with `es.shutdown()` around the use of `tl` here? Right now it is relying on the finalizer of `ThreadPoolExecutor` to cleanup the non-daemon thread.